src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -173,11 +173,23 @@ public StreamingResponseBody streamEvents(
         };
     }
 
+    private String invalidCursorMessage(final CursorError error, final Cursor cursor) {
+        switch (error) {
+            case PARTITION_NOT_FOUND:
+                return "non existing partition " + cursor.getPartition();
+            case EMPTY_PARTITION:
+                return "partition " + cursor.getPartition() + " is empty";
+            case UNAVAILABLE:
+                return "offset " + cursor.getOffset() + " for partition " + cursor.getPartition() + " unavailable";
```

Comment:
Sure. But I did some research and it seems that "is unavailable" could be better since it emphasises the transience of the state. It means that it's unavailable now, but could had been available in the past or even might be available in the future. So I changed it 👍